combine_FOGannotations.m

```diff
@@ -23,277 +23,469 @@ function combine_FOGannotations(filename_rater1, filename_rater2, filename_combi
-tolerance=tolerance_sec*sf;  
- warning('OFF', 'MATLAB:table:ModifiedAndSavedVarnames')
-
-%% collect the annotations of these files
-annotations{1}=readtable(filename_rater1, 'ReadVariableNames', 1, 'HeaderLines', 0);
-annotations{2}=readtable(filename_rater2, 'ReadVariableNames', 1, 'HeaderLines', 0);
+sf=1000;    % choose sampling frequency
+ts=(1/sf);  % time steps
+tolerance=tolerance_sec*sf;
+warning('OFF', 'MATLAB:table:ModifiedAndSavedVarnames'); % Surpress error notification
+
+opts_rater2 = detectImportOptions(filename_rater2);         % read the import options
+opts_rater2 = setvartype(opts_rater2,{'FOG_Trigger','FOG_Type'},'char');   % set to char, default is otherwise double
+opts_rater1 = detectImportOptions(filename_rater1);
+opts_rater1 = setvartype(opts_rater1,{'FOG_Trigger','FOG_Type'},'char');
+% Read annotations
+annotations{1}=CleanAnn(readtable(filename_rater1, opts_rater1));
+annotations{2}=CleanAnn(readtable(filename_rater2, opts_rater2));
+
+%%
 for i=1:2
-  % only select FOG annotations & notes
-  FOG_annotations{i}=annotations{i}(~ismissing(annotations{i}.FOG_Trigger(:))|~ismissing(annotations{i}.FOG_Trigger(:)),:);
-  NOTES{i}=annotations{i}(~ismissing(annotations{i}.NOTES(:)) & (ismissing(annotations{i}.FOG_Trigger(:))&ismissing(annotations{i}.FOG_Type(:))),:); % extra notes that were not annotated during a FOG
-  % check if each FOG has been labeled with a FOG_Trigger and a FOG_Type
-  idx=find(ismissing(FOG_annotations{i}.FOG_Trigger(:))|ismissing(FOG_annotations{i}.FOG_Type));
-  if ~isempty(idx)
-    warning('Not all FOG events were both annotated for FOG_Trigger and FOG_Type for rater %.0d', i)
-    display(FOG_annotations{i}(idx,:))
-  end
-  % calculate total duration based on the gait tasks
-  gait_tasks{i}=annotations{i}(~ismissing(annotations{i}.gait_task(:)),:);
-  if isempty(gait_tasks{i})
-    warning('No gait_tasks were found for rater %.0d. Assuming that the gait_task ended after the last FOG event.', i)
-    duration_gait_tasks{i}=max(FOG_annotations{i}.EndTime_Ss_msec);
-  else
-    duration_gait_tasks{i}=sum(gait_tasks{i}.EndTime_Ss_msec-gait_tasks{i}.BeginTime_Ss_msec);
-  end
+    % only select FOG annotations & notes
+    FOG_annotations{i}=annotations{i}(~ismissing(annotations{i}.FOG_Trigger(:))|~ismissing(annotations{i}.FOG_Trigger(:)),:);
+    % check if each FOG has been labeled with a FOG_Trigger and a FOG_Type
+    idx=find(ismissing(FOG_annotations{i}.FOG_Trigger(:))|ismissing(FOG_annotations{i}.FOG_Type));
+    if ~isempty(idx)
+        warning('Not all FOG events were both annotated for FOG_Trigger and FOG_Type for rater %.0d', i)
+        display(FOG_annotations{i}(idx,:))
+    end
+    
+    % calculate total duration based on the gait tasks
+    gait_tasks{i}=annotations{i}(~ismissing(annotations{i}.gait_task(:)),:);
+    if isempty(gait_tasks{i})
+        warning('No gait_tasks were found for rater %.0d. Assuming that the gait_task ended after the last FOG event.', i)
```

Comment:
never make duration gait task dependent on last FOG event; at least one of raters should have gait_task, otherwise throw error
(check code Emilie for extra tiers: check if both annotations are exactly the same)

Reply:
one reason to leave it this way is for people only using the algorithm for comparing the FOG events but not calculating the agreement parameters.